cpp/include/cuml/common/callback.hpp

```diff
@@ -42,7 +42,7 @@ class GraphBasedDimRedCallback : public Callback {
 
  protected:
   int n;
-  int n_components;
+  uint64_t n_components;
```

Comment:
Let's make sure to include `stdint.h` or (even better!) switch this to `std::uint64_t` and include `<cstdint>`